src/sentry/static/sentry/app/components/button.tsx

```diff
@@ -358,7 +358,7 @@ const getIconMargin = ({size, hasChildren}: IconProps) => {
   return size && size.endsWith('small') ? '6px' : '8px';
 };
 
-const Icon = styled('span')<IconProps>`
+const Icon = styled('span')<IconProps & Omit<StyledButtonProps, 'theme'>>`
```

Comment:
I wouldn't do that because the IconProps type already has everything that is needed. To fix the type issue, I would prefer to update the getFontSize method. As below:

`const getFontSize = ({size, theme}: Pick<StyledButtonProps, 'size' | 'theme'>) => {`